### ISSUE TITLE
question: recipes and cross-validation

### ISSUE THREAD

**Issue by ericlecoutre** [topepo/caret#995]
Question: is this correct that when using cross-validation, recipes are **not** prepared on each fold but on whole train? 
If not, I would simply display a message at the beginning of every fold.
If so, I would add a parameter to eventually allow preparing/learning recipe on individual folds. 
This is especially important for `embed` functions for which in a sense we already build a part of the model

**Comment by ericlecoutre** [comment]
For the selected champion model, which one is used? The one associated to the fold leading to champion model or the final one at entire training level?

**Comment by topepo** [comment]
Once the tuning parameters are selected, the recipe is created on the entire training set and the final model (with the optimal parameter set) is created on the recipe results from the entire training set. 

[This issue](https://github.com/topepo/FES/issues/50) might also help to explain things.

**Comment by Tirgit** [comment]
Hi Max - Could you clarify this a but further - when using a given algorithm and specifying a grid, I am curious about the output model (by model I mean e.g. the coefficients for each features that will be used to make a prediction in a hold-out test set):

a) Is the model that is the output from this line the BEST model with the lowest error amongst the validation sets in the cv framework? I.e. if the grid search is run with 5 fold CV, then the output model would be ONE of FIVE possible models with e.g. the highest ROC AUC or whatever specified?

b) All validations splits are assessed and their error metrics are averaged, the best performing C from the grid is determined and subsequently, this C is run on ALL training data and the resulting model that is the output? I.e. using the above example, for C=1 and C=10 in the grid, the five and five ROC AUC results will be averaged and if C=10 is the winner, C=10 will be used to re-run the model on all training data not using CV and that model is the output?

Thank you so much for your time and thanks for all your work on caret!

**Comment by topepo** [comment]
@Tirgit 

Let's say that you are doing KNN with `k = 11:13` and 5-fold CV. You end up fitting 15 models and then average the results of the 5 models. This gives you three averages (let's say that they are accuracy values). By default, `train` picks the value of `k` with the largest accuracy. 

At this point, all of the 15 models are discarded and a 16th model is with with the best `k` and the entire training set.
